core/app/modules/kit-library/data/repository.php

```diff
@@ -104,6 +125,60 @@ public function get_download_link( $id ) {
 		return [ 'download_link' => $response->download_link ];
 	}
 
+	/**
+	 * @param $user_id
+	 * @param $id
+	 *
+	 * @return array
+	 * @throws Wp_Error_Exception
+	 */
+	public function add_to_favorites( $user_id, $id ) {
+		// To check that the kit is exists.
+		$kit = $this->find( $id, $user_id, [
+			'manifest_included' => false,
+		] );
+
+		$favorites = $this->get_user_favorites_meta( $user_id );
+
+		if ( in_array( $kit['id'], $favorites, true ) ) {
+			return $kit;
+		}
+
+		$favorites[] = $kit['id'];
+
+		$this->save_user_favorites_meta( $user_id, $favorites );
+
+		$kit['is_favorite'] = true;
+
+		return $kit;
+	}
+
+	/**
+	 * @param $user_id
+	 * @param $id
+	 *
+	 * @return array
+	 * @throws Wp_Error_Exception
+	 */
+	public function remove_from_favorites( $user_id, $id ) {
+		// To check that the kit is exists.
+		$kit = $this->find( $id, $user_id, [
+			'manifest_included' => false,
+		] );
+
+		$favorites = $this->get_user_favorites_meta( $user_id );
+
+		$favorites = array_filter( $favorites, function ( $item ) use ( $kit ) {
+			return $item !== $kit['id'];
+		} );
+
+		$this->save_user_favorites_meta( $user_id, $favorites );
+
+		$kit['is_favorite'] = false;
+
+		return $kit;
+	}
```

Comment:
Can you re-use the similar existing method?
https://github.com/elementor/elementor/blob/31629fa3b3b45b58356208102016a8406d21a7eb/includes/template-library/sources/base.php#L190-L204